### PR TITLE
[docs] descriptive comments of the decoder C++ api

### PR DESCRIPTION
**Pull request by bjuncek**
Adding a more descriptive commenting of parts of the decoder API in an effort to move the documentation forward. No actually code has been changed.

### CODE REVIEW

**Comment by fmassa** [comment]
FYI clang-format is failing

**Comment by bjuncek** [comment]
Clang is now passing; @prabhat00155 could you give it a pass for a merge?

File: torchvision/csrc/io/decoder/decoder.cpp [hunk 1]
```diff
@@ -213,6 +213,12 @@ Decoder::~Decoder() {
   cleanUp();
 }
 
+// Initialise the format context that holds information about the container and
+// fill it with minimal information about the format (codecs are not opened
+// here). Function reads in intformation about the streams from the container
+// into inputCtx and then passes it to decoder::openStreams. Finally, if seek is
+// specified, it seeks into the correct frame (note, the seek defined here is
```
[the remark this answers is not in the document]
Comment (bjuncek) [reply torchvision/csrc/io/decoder/decoder.cpp:220]: There is a flag for fast seek implementation within decoder metadata. You can see the implementation (and we should try fbsync it once we're back from vacation here) https://github.com/pytorch/vision/pull/3179

File: torchvision/csrc/io/decoder/decoder.cpp [hunk 2]
```diff
@@ -379,7 +385,7 @@ bool Decoder::init(
     cleanUp();
     return false;
   }
-
+  // not sure what this does?
```
[the remark this answers is not in the document]
Comment (bjuncek) [reply torchvision/csrc/io/decoder/decoder.cpp:388]: Thanks!

**Pull request merged**